python/dgl/sampling/neighbor.py

```diff
@@ -99,15 +96,19 @@ def sample_etype_neighbors(g, nodes, etype_field, fanout, edge_dir='in', prob=No
     if etype_field not in g.edata:
         raise DGLError("The graph should have {} in the edge data" \
                        "representing the edge type.".format(etype_field))
-    if isinstance(fanout, int) is False:
-        raise DGLError("The fanout should be an integer")
-    if isinstance(nodes, dict) is True:
+    # (BarclayII) because the homogenized graph no longer contains the *name* of edge
+    # types, the fanout argument can no longer be a dict of etypes and ints, as opposed
+    # to sample_neighbors.
+    if not F.is_tensor(fanout):
```

Comment:
Can this be a list of int?

Reply:
This function is not being used in user interface (it is only called via NeighborSampler, where it expects a dict of fanouts, as shown in the distributed data loader unit tests).  I also need to convert the list into a tensor for C++ anyway.